### PR TITLE
Combine db tests

### PR DESCRIPTION
**Pull request by rnorth**
This change relocates all (non-proprietary) DB tests into a single module, to reduce the amount of boilerplate. It is envisaged that we can clean up these tests further going forward.

Additionally this PR includes changes to reduce the initial memory consumption of MySQL and MariaDB containers for reliability reasons.

### CODE REVIEW

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import static org.rnorth.visibleassertions.VisibleAssertions.assertEquals;
+
+@RunWith(Parameterized.class)
+public class JDBCDriverTest {
+
+    @Parameterized.Parameters(name = "{index} - {0}")
+    public static Iterable<Object[]> data() {
+        return asList(
+                new Object[][]{
+                        {"jdbc:tc:mysql:5.5.43://hostname/databasename", false, false, false},
```
Comment (bsideup) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: awesome!

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                        {"jdbc:tc:mariadb://hostname/databasename", false, false, false},
+                        {"jdbc:tc:mariadb://hostname/databasename?useUnicode=yes&characterEncoding=utf8", false, true, false},
+                        {"jdbc:tc:mariadb://hostname/databasename?TC_INITSCRIPT=somepath/init_mariadb.sql", true, false, false},
+                        {"jdbc:tc:mariadb://hostname/databasename?TC_INITFUNCTION=org.testcontainers.jdbc.JDBCDriverTest::sampleInitFunction", true, false, false},
+                        {"jdbc:tc:mysql:5.6://hostname/databasename?TC_MY_CNF=somepath/mysql_conf_override", false, false, true},
+                        {"jdbc:tc:mariadb:10.1.16://hostname/databasename?TC_MY_CNF=somepath/mariadb_conf_override", false, false, true}
+                });
+    }
+
+    @Parameterized.Parameter(0)
```
Comment (bsideup) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: maybe static import `Parameter`?

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        });
+        dataSource.close();
+    }
+
+    private void performTestForScriptedSchema(String jdbcUrl) throws SQLException {
+        HikariDataSource dataSource = getDataSource(jdbcUrl, 1);
+        new QueryRunner(dataSource).query("SELECT foo FROM bar WHERE foo LIKE '%world'", (ResultSetHandler<Object>) rs -> {
+            rs.next();
+            String resultSetString = rs.getString(1);
+            assertEquals("A basic SELECT query succeeds where the schema has been applied from a script", "hello world", resultSetString);
```
Comment (bsideup) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: I'm just wondering what if ResultSetHandler will not be called.

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            return true;
+        });
+        dataSource.close();
```
Comment (bsideup) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: HikariDataSource is AutoClosable, you can use try-with-resources

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java (hunk 2 above)
Comment (rnorth) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: Definitely - will do.

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java (hunk 3 above)
Comment (rnorth) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: Well spotted - will address that!

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java (hunk 4 above)
Comment (rnorth) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: Will do

File: modules/jdbc-test/pom.xml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        <dependency>
+            <groupId>commons-dbutils</groupId>
+            <artifactId>commons-dbutils</artifactId>
+            <version>1.6</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.googlecode.junit-toolbox</groupId>
+            <artifactId>junit-toolbox</artifactId>
```
Comment (rnorth) [inline modules/jdbc-test/pom.xml]: Will remove

**Comment by rnorth** [comment]
As this PR is completely included in #243 but doesn't include some bugfixes that are in the other PR, I'm going to close this one. Seems more straightforward that way.

**Pull request closed without merging**